Remove `id` from paths to instances in style warnings

Mingun opened the pull request:
It seems relatively new error, but the code was not changed since 2021...

Fixes errors like:
```
[info] - style_bad_num_inst_value *** FAILED ***
[info]   [style_bad_num_inst_value.ksy: /instances/size_of_foo/id:
[info]   	warning: use `num_foos` instead of `size_of_foo`, given that it's only used as repeat count of `foos` (see https://doc.kaitai.io/ksy_style_guide.html#attr-id)
[info]   ]
[info]     did not equal
[info]   [style_bad_num_inst_value.ksy: /instances/size_of_foo:
[info]   	warning: use `num_foos` instead of `size_of_foo`, given that it's only used as repeat count of `foos` (see https://doc.kaitai.io/ksy_style_guide.html#attr-id)
[info]   ] (SimpleMatchers.scala:34)
```

`/instances/size_of_foo/id` is not-existing path. Instances does not have `id`, the name of instance is its `id`.

Fixes https://github.com/kaitai-io/kaitai_struct/issues/920.

Also fixes a couple of other errors

Mingun committed:
Remove `id` from paths to instances in style warnings

Fixes https://github.com/kaitai-io/kaitai_struct/issues/920

Error count: 59 -> 55 (-4). Fixed:

[info] - style_bad_len_inst_pos *** FAILED ***
[info]   [style_bad_len_inst_pos.ksy: /instances/size_of_foo/id:
[info]          warning: use `len_foo` instead of `size_of_foo`, given that it's only used as a byte size of `foo` (see https://doc.kaitai.io/ksy_style_guide.html#attr-id)
[info]   ]
[info]     did not equal
[info]   [style_bad_len_inst_pos.ksy: /instances/size_of_foo:
[info]          warning: use `len_foo` instead of `size_of_foo`, given that it's only used as a byte size of `foo` (see https://doc.kaitai.io/ksy_style_guide.html#attr-id)
[info]   ] (SimpleMatchers.scala:34)
[info] - style_bad_len_inst_value *** FAILED ***
[info]   [style_bad_len_inst_value.ksy: /instances/size_of_foo/id:
[info]          warning: use `len_foo` instead of `size_of_foo`, given that it's only used as a byte size of `foo` (see https://doc.kaitai.io/ksy_style_guide.html#attr-id)
[info]   ]
[info]     did not equal
[info]   [style_bad_len_inst_value.ksy: /instances/size_of_foo:
[info]          warning: use `len_foo` instead of `size_of_foo`, given that it's only used as a byte size of `foo` (see https://doc.kaitai.io/ksy_style_guide.html#attr-id)
[info]   ] (SimpleMatchers.scala:34)
[info] - style_bad_num_inst_pos *** FAILED ***
[info]   [style_bad_num_inst_pos.ksy: /instances/size_of_foo/id:
[info]          warning: use `num_foos` instead of `size_of_foo`, given that it's only used as repeat count of `foos` (see https://doc.kaitai.io/ksy_style_guide.html#attr-id)
[info]   ]
[info]     did not equal
[info]   [style_bad_num_inst_pos.ksy: /instances/size_of_foo:
[info]          warning: use `num_foos` instead of `size_of_foo`, given that it's only used as repeat count of `foos` (see https://doc.kaitai.io/ksy_style_guide.html#attr-id)
[info]   ] (SimpleMatchers.scala:34)
[info] - style_bad_num_inst_value *** FAILED ***
[info]   [style_bad_num_inst_value.ksy: /instances/size_of_foo/id:
[info]          warning: use `num_foos` instead of `size_of_foo`, given that it's only used as repeat count of `foos` (see https://doc.kaitai.io/ksy_style_guide.html#attr-id)
[info]   ]
[info]     did not equal
[info]   [style_bad_num_inst_value.ksy: /instances/size_of_foo:
[info]          warning: use `num_foos` instead of `size_of_foo`, given that it's only used as repeat count of `foos` (see https://doc.kaitai.io/ksy_style_guide.html#attr-id)
[info]   ] (SimpleMatchers.scala:34)

generalmimon commented:
@Mingun:

> It seems relatively new error

It is. I recently extended the `formats_err` test suite: https://github.com/kaitai-io/kaitai_struct_tests/compare/c5f366f4ce108650d585f9218deb9de345c44e7f...0427d48a0da213241f75f7029cd08217d0e1c7ef

The test you're fixing here was added in https://github.com/kaitai-io/kaitai_struct_tests/commit/c911e0a302939208d796d344294d0caa026b6941 to demonstrate https://github.com/kaitai-io/kaitai_struct/issues/920. So ideally you should add `Fixes https://github.com/kaitai-io/kaitai_struct/issues/920` to the PR description and commit message.

Mingun committed:
Fix paths in "duplicate attribute ID '$ID', previously defined at $PATH" errors

Error count: 55 -> 48 (-7). Fixed:

[info] - id_clash_params_vs_inst_pos *** FAILED ***
[info]   [id_clash_params_vs_inst_pos.ksy: /instances/foo:
[info]          error: duplicate attribute ID 'foo', previously defined at /params/0
[info]   ]
[info]     did not equal
[info]   [id_clash_params_vs_inst_pos.ksy: /instances/foo:
[info]          error: duplicate attribute ID 'foo', previously defined at /params/0/id
[info]   ] (SimpleMatchers.scala:34)
[info] - id_clash_params_vs_inst_value *** FAILED ***
[info]   [id_clash_params_vs_inst_value.ksy: /instances/bar:
[info]          error: duplicate attribute ID 'bar', previously defined at /params/1
[info]   ]
[info]     did not equal
[info]   [id_clash_params_vs_inst_value.ksy: /instances/bar:
[info]          error: duplicate attribute ID 'bar', previously defined at /params/1/id
[info]   ] (SimpleMatchers.scala:34)
[info] - id_clash_params_vs_seq *** FAILED ***
[info]   [id_clash_params_vs_seq.ksy: /seq/0:
[info]          error: duplicate attribute ID 'foo', previously defined at /params/1
[info]   ]
[info]     did not equal
[info]   [id_clash_params_vs_seq.ksy: /seq/0/id:
[info]          error: duplicate attribute ID 'foo', previously defined at /params/1/id
[info]   ] (SimpleMatchers.scala:34)
[info] - id_clash_seq_vs_inst_pos *** FAILED ***
[info]   [id_clash_seq_vs_inst_pos.ksy: /instances/foo:
[info]          error: duplicate attribute ID 'foo', previously defined at /seq/0
[info]   ]
[info]     did not equal
[info]   [id_clash_seq_vs_inst_pos.ksy: /instances/foo:
[info]          error: duplicate attribute ID 'foo', previously defined at /seq/0/id
[info]   ] (SimpleMatchers.scala:34)
[info] - id_clash_seq_vs_inst_value *** FAILED ***
[info]   [id_clash_seq_vs_inst_value.ksy: /instances/bar:
[info]          error: duplicate attribute ID 'bar', previously defined at /seq/1
[info]   ]
[info]     did not equal
[info]   [id_clash_seq_vs_inst_value.ksy: /instances/bar:
[info]          error: duplicate attribute ID 'bar', previously defined at /seq/1/id
[info]   ] (SimpleMatchers.scala:34)
[info] - id_dup_params *** FAILED ***
[info]   [id_dup_params.ksy: /params/2:
[info]          error: duplicate attribute ID 'foo', previously defined at /params/0
[info]   ]
[info]     did not equal
[info]   [id_dup_params.ksy: /params/2/id:
[info]          error: duplicate attribute ID 'foo', previously defined at /params/0/id
[info]   ] (SimpleMatchers.scala:34)
[info] - id_dup_seq *** FAILED ***
[info]   [id_dup_seq.ksy: /seq/2:
[info]          error: duplicate attribute ID 'foo', previously defined at /seq/0
[info]   ]
[info]     did not equal
[info]   [id_dup_seq.ksy: /seq/2/id:
[info]          error: duplicate attribute ID 'foo', previously defined at /seq/0/id
[info]   ] (SimpleMatchers.scala:34)

Mingun committed:
Use concatenation with one element instead of list with one element

Mingun committed:
Fix duplication of `type` in the path of `switch-on` types in errors

Error count: 48 -> 47 (-1). Fixed:

[info] - type_unknown_switch *** FAILED ***
[info]   [type_unknown_switch.ksy: /seq/0/type/cases/IntNum(42)/type:
[info]          error: unable to find type 'some_unknown_name', searching from type_unknown_switch
[info]   ]
[info]     did not equal
[info]   [type_unknown_switch.ksy: /seq/0/type/cases/IntNum(42):
[info]          error: unable to find type 'some_unknown_name', searching from type_unknown_switch
[info]   ] (SimpleMatchers.scala:34)

Mingun commented:
@generalmimon, could you look again? This PR fixes 13 errors

Mingun commented:
@GreyCat, @generalmimon , this PR reduces error count from currently 54 to 42. It is small win, do you mind to merge it?

GreyCat reviewed (approved):
Generally, looks good to me and relatively straightforward fix! Thanks @Mingun!

@generalmimon, do you have any concerns or shall we merge this?

generalmimon reviewed (approved):
I think it's fine too, it indeed fixes some of the test failures.

You know I'm not a big fan of adding comments unless it's absolutely necessary, but if you do, I'd prefer them to make sense...
